🐛 Overwrite Bulkrax::ScheduleRelationshipsJob

kirkkwang opened the pull request:
This commit will use a different query to get the pending_num value.  We were seeing that the query in current Bulkrax is prematurely going to zero. Testing it out in UTK and refining it before considering contributing back to Bulkrax.

Ref:
- https://github.com/notch8/utk-hyku/issues/737

Commit:
🐛 Overwrite Bulkrax::ScheduleRelationshipsJob

This commit will use a different query to get the pending_num value.  We
were seeing that the query in current Bulkrax is prematurely going to
zero. Testing it out in UTK and refining it before considering
contributing back to Bulkrax.

Ref:
- https://github.com/notch8/utk-hyku/issues/737

bkiahstroud commented on app/jobs/bulkrax/schedule_relationships_job_decorator.rb:
```diff
@@ -0,0 +1,29 @@
+        Bulkrax::Status
+        .where(statusable_id: importer.entries.pluck(:id))
+        .where(runnable_id: importer.last_run.id)
+        .where(status_message: ['Complete', 'Failed', 'Skipped'])
```
I'm not positive off the top of my head, but I don't think this encompasses all possible statuses. Either way, what we're truly after is "entries that are not pending", so I recommend something like: 

```suggestion
        .where.not(status_message: 'Pending')
```
It may actually be `.where.not(status_message: nil)` but you get the idea

kirkkwang replied:
Thanks sounds good to me, committing it.

kirkkwang committed:
Look for status that are not 'Pending'

Favor using where not 'Pending' because ultimately that's what we're looking for.

Co-authored-by: Kiah Stroud <32469930+bkiahstroud@users.noreply.github.com>

laritakr reviewed (approved):
This seems to be a reasonable query. I know that Kait has work underway regarding statuses & error message handling, so overriding it seems the way to go for the time being.
